src/server/protocol.ts

```diff
@@ -3217,7 +3217,7 @@ namespace ts.server.protocol {
          * values, with insertion text to replace preceding `.` tokens with `?.`.
          */
         readonly includeAutomaticOptionalChainCompletions?: boolean;
-        readonly importModuleSpecifierPreference?: "auto" | "relative" | "non-relative";
+        readonly importModuleSpecifierPreference?: "shortest" | "external-non-relative" | "relative" | "non-relative";

```

Comment:
Kind of nervous that `auto` is going away; can you explain?

Reply:
I’ve renamed `auto` to `shortest` to be more descriptive. Previously, `auto` was an acceptable name because it would pick between the other options (relative and non-relative) based on some heuristic (number of path separators). Now, there are _two_ options that pick between relative and non-relative based on some heuristic. `auto` becomes a bad name because it no longer encompasses _all_ of the other options, it’s no more or less automatic than the new mode, and its name conveys nothing about the heuristic it uses. Values of `"auto"` will still map to `"shortest"`, by merit of it being the default.

Reply:
💯